### DISCUSSION TITLE
Disable Password Confirmation

### DISCUSSION THREAD

**Discussion by eyalrin** [keycloak/keycloak#22360] [Q&A]
Hi all!

I wanted to know if Is there a built-in way in Keycloak to disable password confirmation field? Without the need to create an SPI and/or theme? I couldn't find such a setting in the UI...

**Comment by redarccoder** [comment]
Depends on how you're creating your users. If you're using the REST API the it's as easy as to open the network monitor in your web browser, disable if for one user and then replicate the request while creating the rest of them.

If you're using LDAP there is a checkbox to enable to trust all the users.

Anything besides that you would need to add more details to your question

**Reply by eyalrin** [reply]
Users are being created via the web login page:
<img width="635" alt="Screenshot 2023-08-13 at 10 51 08" src="https://github.com/keycloak/keycloak/assets/8133764/828531a7-6f47-435e-9ad5-b590c8836ef5">

Mind you that I do want the other email validations to work, just the email confirmation to be disabled.

**Comment by redarccoder** [comment]
> Users are being created via the web login page: <img alt="Screenshot 2023-08-13 at 10 51 08" width="635" src="https://user-images.githubusercontent.com/8133764/260289242-828531a7-6f47-435e-9ad5-b590c8836ef5.png">
> 
> Mind you that I do want the other email validations to work, just the email confirmation to be disabled.

The "Confirm Password" is part o the form, not really necessary for Keycloak, you can take a look at Server Development:
https://www.keycloak.org/docs/latest/server_development/#_themes

**Reply by eyalrin** [reply]
Are you sure? because as part of the registration flow there is a step called **Password Validation** - it controls all validations regarding password, including password confirmation. If we disable this step, no pass confirmation is needed but also all other password validations are disabled...
